#! /bin/sh /usr/share/dpatch/dpatch-run
## 12_fix_line_width.dpatch by Goedson Teixeira Paixao <goedson@debian.org>
##
## DP: Replaces the contents of the line width entry
## DP: with the currently set line width if the value
## DP: entered by the user is not valid

@DPATCH@
diff -urNad gpaint-0.3.1~/src/menu.c gpaint-0.3.1/src/menu.c
--- gpaint-0.3.1~/src/menu.c	2007-08-18 18:59:06.000000000 -0300
+++ gpaint-0.3.1/src/menu.c	2007-08-18 18:59:24.000000000 -0300
@@ -292,6 +292,7 @@
     GdkGCValues gcvalues;
     gpaint_canvas *canvas;
     gpaint_drawing *drawing;
+    int position = 0;
    
     debug_fn();
     canvas = canvas_lookup(GTK_WIDGET(editable));  
@@ -299,15 +300,26 @@
     
     tmp = gtk_editable_get_chars(editable, 0, -1);
     g_assert(tmp);
-    sscanf(tmp, "%d", &t);
-    g_free(tmp);
-    g_assert(t != 0);
-   
-    gdk_gc_get_values(drawing->gc, &gcvalues);
-    gdk_gc_set_line_attributes(drawing->gc, t, gcvalues.line_style, gcvalues.cap_style, gcvalues.join_style);
+    if (strlen(tmp) > 0) {
+        sscanf(tmp, "%d", &t);
+        g_free(tmp);
+        if (t > 0) {
+            gdk_gc_get_values(drawing->gc, &gcvalues);
+            gdk_gc_set_line_attributes(drawing->gc, t, gcvalues.line_style, gcvalues.cap_style, gcvalues.join_style);
 
-    /* force the line width entry widget to give up focus */
-    gtk_widget_grab_focus(GTK_WIDGET(canvas->drawing_area));
+            /* force the line width entry widget to give up focus */
+            gtk_widget_grab_focus(GTK_WIDGET(canvas->drawing_area));
+        } else {
+            gdk_gc_get_values(drawing->gc, &gcvalues);
+            tmp = g_strdup_printf("%d", gcvalues.line_width);
+            gtk_editable_delete_text(editable, 0, -1);
+            gtk_editable_insert_text(editable, tmp, strlen(tmp), &position);
+            gtk_editable_set_position(editable, -1);
+            g_free(tmp);
+        }
+    } else {
+      g_free(tmp);
+    }
 }
 
 void
